gperf generated files include absolute path to Xcode

emw-apple opened the pull request:
#### acbbcc88d48b140a185ec4b9302840a1879cfafe
<pre>
gperf generated files include absolute path to Xcode
<a href="https://bugs.webkit.org/show_bug.cgi?id=258634">https://bugs.webkit.org/show_bug.cgi?id=258634</a>
rdar://111130168

Reviewed by Alexey Proskuryakov.

Invoke gperf through a symlink, so that it&apos;s argv[0] doesn&apos;t contain an
absolute path. This works towards making WebCore&apos;s generated sources
fully reproducible across machines.

* Source/WebCore/DerivedSources.make: Always use $(GPERF) as the path to
  gperf, instead of sometimes relying on PATH-based lookup.
* Source/WebCore/Scripts/generate-derived-sources.sh: Create a ./gperf
  symlink in DerivedSources directory, similar to what we already do for
  SRCROOT.

Canonical link: <a href="https://commits.webkit.org/265608@main">https://commits.webkit.org/265608@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/edc54b04401914cb1418416b80129fc89956e079

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11389 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11594 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11940 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13034 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10841 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11410 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13976 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11576 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13759 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11551 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12424 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/9640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13453 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9716 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/10330 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10787 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10484 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13694 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10894 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8967 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10067 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2729 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14340 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10749 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebCore/Scripts/generate-derived-sources.sh:
```diff
@@ -17,7 +17,8 @@ if [ ! "$CC" ]; then
 
 if [ ! "$GPERF" ]; then
-    export GPERF="`xcrun -find gperf`"
+    /bin/ln -sfh "`xcrun -find gperf`" gperf
```
Maybe worth adding a comment here about why. Not everyone will do a git blame.

emw-apple committed:
gperf generated files include absolute path to Xcode
https://bugs.webkit.org/show_bug.cgi?id=258634
rdar://111130168

Reviewed by Alexey Proskuryakov.

Invoke gperf through a symlink, so that it's argv[0] doesn't contain an
absolute path. This works towards making WebCore's generated sources
fully reproducible across machines.

* Source/WebCore/DerivedSources.make: Always use $(GPERF) as the path to
  gperf, instead of sometimes relying on PATH-based lookup.
* Source/WebCore/Scripts/generate-derived-sources.sh: Create a ./gperf
  symlink in DerivedSources directory, similar to what we already do for
  SRCROOT.

Canonical link: https://commits.webkit.org/265608@main
